book/setting_up.md

```diff
@@ -45,14 +45,14 @@ Our `main.cpp` is still empty, lets rectify that by adding in a [button].
 ```cpp
 // gui/main.cpp
 
-#include <QtWidgets/QApplication>
 #include <QtWidgets/QPushButton>
+#include <QtWidgets/QApplication>
 
 int main(int argc, char **argv) {
   QApplication app(argc, argv);
 
-  QPushButton button("Hello World");
-  button.show();
+  QPushButton *button = new QPushButton("Hello World");
```

Comment:
Do we need to delete this guy afterwards? We might be able to use a `unique_ptr` to automatically handle lifetimes, as well as show off more "modern" C++.

Reply:
Ah, indeed, I was confused by recently only using widgets/mainwindows which have `Qt::WA_DeleteOnClose` set. Looking quickly at the latest Qt tutorials, they put all toplevel windows on the stack now. So for consistency and code simpliciity I would propose to follow them there. [Patch update done next]